pkg/apply/desiredset_process_test.go

```diff
@@ -109,3 +113,157 @@ func Test_multiNamespaceList(t *testing.T) {
 		})
 	}
 }
+
+func Test_canIndexByHash(t *testing.T) {
+	const hash = "somehash"
+	selector, err := GetSelector(map[string]string{LabelHash: hash})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type args struct {
+		indexer  cache.Indexer
+		selector labels.Selector
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantHash string
+		want     bool
+	}{
+		{name: "indexer not configured", args: args{
+			indexer:  cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{}),
+			selector: selector,
+		}, wantHash: "", want: false},
+		{name: "indexer configured", args: args{
+			indexer: cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{byHash: func(obj interface{}) ([]string, error) {
+				return nil, nil
+			}}),
+			selector: selector,
+		}, wantHash: hash, want: true},
+		{name: "non exact hash selector", args: args{
+			indexer: cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{byHash: func(obj interface{}) ([]string, error) {
+				return nil, nil
+			}}),
+			selector: labels.Everything(),
```

Comment:
nit:
labels.Everything() does not have any Requirements so it fails early before checking if the selector has the desired hash key.

Reply:
Nice catch, thanks!